control/standard_commands.go

```diff
@@ -0,0 +1,38 @@
+package control
+
+// StandardCommand - Data type to store standard command data
+type StandardCommand struct {
+	robotID         int
+	linearVelocity  float64
+	angularVelocity float64
+}
+
+// RobotID returns the robot id of the command
+func (sd StandardCommand) RobotID() int {
+	return sd.robotID
+}
+
+// LinearVelocity returns the linear velocity of the command
+func (sd StandardCommand) LinearVelocity() float64 {
+	return sd.linearVelocity
+}
+
+// AngularVelocity returns the angular velocity of the command
+func (sd StandardCommand) AngularVelocity() float64 {
+	return sd.angularVelocity
+}
+
+// SetRobotID sets the robot id of the command
+func (sd *StandardCommand) SetRobotID(id int) {
```

Comment:
Recomendo tratar isso com campos exportados e não com funções _Set_. O pipeline ficará melhor se o comando for construído e enviado, menos problemas de concorrência, e na construção tu passa os parâmetros no construtor, ou seta o campo exportado, como se fosse uma mensagem mesmo. Se não tu vai criar e chamar _Set_ de tudo. E se a ideia é ter uma instância única de comando pra cada robo e justificar o uso do _Set_, acho que vamos trocar a facilidade do outro modelo (que é um pouquinho só mais caro) mas que vai compensar se tivermos que tratar a concorrência com _mutex_ e etc.

Reply:
Concordo. Não entendi o que tu falou do outro modelo. É em referência ao modelo mundo?

Reply:
Não não. Outro modelo é o de _Data Flow_, tu cria o comando e envia ele pra onde for (_CommandsRepository_, por exemplo), era a isso que me referia, ao invés de ter o repositorio segurando instâncias que ficam sendo alteradas _in place_ no tempo.

Reply:
Entendido.